compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1607,6 +1607,18 @@ rustc_queries! {
         desc { "looking at the source for a crate" }
         separate_provide_extern
     }
+    /// Returns the debugger visualizers defined in this crate.
+    query get_debugger_visualizers(_: ()) -> Vec<rustc_span::DebuggerVisualizerFile> {
+        storage(ArenaCacheSelector<'tcx>)
+        desc { "looking up the debugger visualizers of the current crate" }
+        separate_provide_extern
+    }
+    /// Returns the debugger visualizers defined in another crate by loading it from metadata.
+    query used_crate_debugger_visualizers(_: CrateNum) -> Vec<rustc_span::DebuggerVisualizerFile> {
+        storage(ArenaCacheSelector<'tcx>)
+        desc { "looking up the debugger visualizers of a linked crate" }
+        separate_provide_extern
+    }
```

Comment:
These queries work in separate ways. The first query looks at the AST and pulls out all of the `#[debugger_visualizer]` attributes to create a `DebuggerVisualizerFile`. While the second query loads the debugger visualizers defined in other crates from the crate metadata. I think it would be best to keep them separate. Is there a reason to combine them?